Tasks/PerformActionOnMachineGroup/AzureHelper.ps1

```diff
@@ -71,26 +71,41 @@ function Initialize-AzureHelper
         $providerDataName = $machineGroup.ProviderDataList[0].Name
         Write-Verbose "Getting providerData : $providerDataName" -Verbose
         $providerData = Get-ProviderData -ProviderDataName $providerDataName -Connection $connection
-        $subscriptionName = $providerData.Properties.GetProperty("SubscriptionName")     
-        $username = $providerData.Properties.GetProperty("Username")
-        $password = $providerData.Properties.GetProperty("Password")
-
-        if( ![string]::IsNullOrEmpty($subscriptionName) -and ![string]::IsNullOrEmpty($username) -and ![string]::IsNullOrEmpty($password) )
+        $subscriptionId = $providerData.Properties.GetProperty("SubscriptionId")     
```

Comment:
There is a dll called Dtl.common where all the constants are defined. Please move "SubscriptionId" there